src/planner/match/Expand.cpp

```diff
@@ -199,19 +203,27 @@ Status Expand::expandStep(const EdgeInfo& edge,
             }
             // filter tag
             return new VertexExpression();
-        });
+        };
+        RewriteVisitor visitor(std::move(matcher), std::move(rewriter));
+
         filter->accept(&visitor);
         auto filterNode = Filter::make(matchCtx_->qctx, root, filter);
         filterNode->setColNames(root->colNames());
         root = filterNode;
     }
 
     if (edge.filter != nullptr) {
-        RewriteMatchLabelVisitor visitor([](const Expression* expr) {
+        auto matcher = [](const Expression *expr) -> bool{
+            return expr->kind() == Expression::Kind::kLabel
+                || expr->kind() == Expression::Kind::kLabelAttribute;
+        };
```

Comment:
Why not provide a function to avoid to define these lambda repeatly?

Reply:
ACK.